### PR TITLE
add some tests

### PR DESCRIPTION
**Pull request by Agi-dev**

### CODE REVIEW

File: tests/Itq/Common/Service/AttachmentServiceTest.php [hunk 1]
```diff
@@ -31,24 +31,119 @@ public function s()
...
+            ],
+            '2 - build from base64'            => [
+                ['name' => 'test.pdf', 'base64_content' => base64_encode('Je suis base64')],
+                [],
+                [],
+                ['name' => 'test.pdf', 'type' => 'application/pdf', 'content' => base64_encode('Je suis base64')]
+            ],
+        ];
+    }
+
```
[the remark this answers is not in the document]
Comment (Agi-dev) [reply tests/Itq/Common/Service/AttachmentServiceTest.php]: If i remove blank line : codesniffer said :
FILE: /Users/francois/Workspaces/php/itiQiti/common/tests/Itq/Common/Service/AttachmentServiceTest.php
------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------------
 152 | ERROR | [x] Expected 1 newline at end of file; 0 found

File: src/Itq/Common/Service/AttachmentService.php [hunk 2]
```diff
@@ -73,13 +78,13 @@ protected function buildFromPath(array $definition, array $params = [], array $o
         unset($params);
         unset($options);
 
-        if (!is_file($definition['path'])) {
-            throw $this->createNotFoundException("File '%s' not found", $definition['path']);
-        }
+//        if ($this->getFilesystemService()->isReadableFile($definition['path'])) {
```
Comment (ohoareau) [inline src/Itq/Common/Service/AttachmentService.php]: no commented code. why commented ?

File: tests/Itq/Common/Command/DocGenerateCommandTest.php [hunk 3]
```diff
@@ -70,9 +70,8 @@ public function getRunData()
     {
         return [
             'missing-type-and-path' => [['default', 'doc'], []],
-            'missing-type'          => [['default', 'thepath'], ['path' => 'thepath']],
-            'has-type-and-path'     => [['thetype', 'thepath'], ['path' => 'thepath', 'type' => 'thetype']],
+            'missing-type' => [['default', 'thepath'], ['path' => 'thepath']],
```
Comment (ohoareau) [inline tests/Itq/Common/Command/DocGenerateCommandTest.php]: be careful to "align" the arrows.

File: tests/Itq/Common/Form/EventListener/AddIbanFieldSubscriberTest.php [hunk 4]
```diff
@@ -79,20 +81,20 @@ public function getPreSubmitData()
     {
         return [
             '0 - authorized ibans' => [
-                ['iban'],
+                ['iban', ],
```
Comment (ohoareau) [inline tests/Itq/Common/Form/EventListener/AddIbanFieldSubscriberTest.php]: no trailing comma on one-liner array definition.

File: tests/Itq/Common/Form/EventListener/AddIbanFieldSubscriberTest.php [hunk 5]
```diff
@@ -79,20 +81,20 @@ public function getPreSubmitData()
...
                 'FR89370400440532013000',
                 'FR89370400440532013000',
             ],
             '1 - not authorized ibans' => [
-                ['iban', 'text', ['constraints' => [new Iban(['groups'  => ['create', 'update']])]]],
+                ['iban', 'text', ['constraints' => [new Iban(['groups' => ['create', 'update', ]])]]],
```
Comment (ohoareau) [inline tests/Itq/Common/Form/EventListener/AddIbanFieldSubscriberTest.php]: no trailing comma on one-liner array definition.

File: tests/Itq/Common/Form/EventListener/AddIbanFieldSubscriberTest.php [hunk 6]
```diff
@@ -79,20 +81,20 @@ public function getPreSubmitData()
...
                 'FR89370400440532013000',
                 'FR89370400440532013001',
             ],
             '2 - none' => [
-                ['iban', 'text', ['constraints' => [new Iban(['groups'  => ['create', 'update']])]]],
+                ['iban', 'text', ['constraints' => [new Iban(['groups' => ['create', 'update', ]])]]],
```
Comment (ohoareau) [inline tests/Itq/Common/Form/EventListener/AddIbanFieldSubscriberTest.php]: no trailing comma on one-liner array definition.

File: tests/Itq/Common/Service/AttachmentServiceTest.php [hunk 7]
```diff
@@ -36,19 +36,117 @@ public function s()
...
+     * @param string $filename
+     * @param string $expected
+     *
+     * @group        unit
+     *
+     * @dataProvider getMimeTypeFromFileNameData
+     */
+    public function testGetMimeTypeFromFileName($filename, $expected)
+    {
+        $m = $this->accessible($this->s(), 'getMimeTypeFromFileName');
```
Comment (ohoareau) [inline tests/Itq/Common/Service/AttachmentServiceTest.php]: $m is used only once, so can avoid using a local variable and put everything on one line.

File: tests/Itq/Common/Service/BatchServiceTest.php [hunk 8]
```diff
@@ -12,20 +12,65 @@
...
     /**
+     * @return Service\BatchService
+     */
+    public function s()
+    {
+        /** @noinspection PhpIncompatibleReturnTypeInspection */
+
+        return parent::s();
+    }
+
```
Comment (ohoareau) [inline tests/Itq/Common/Service/BatchServiceTest.php]: remove blank lines between methods.

File: tests/Itq/Common/Service/BatchServiceTest.php [hunk 9]
```diff
@@ -12,20 +12,65 @@
...
+    /**
      * @return array
      */
     public function constructor()
     {
         return [$this->mockedEventDispatcher()];
     }
+
```
Comment (ohoareau) [inline tests/Itq/Common/Service/BatchServiceTest.php]: remove blank lines between methods.

File: tests/Itq/Common/Service/BatchServiceTest.php [hunk 10]
```diff
@@ -12,20 +12,65 @@
...
+    /**
+     * @group unit
+     */
+    public function testExecuteWithUnknownBatchThrowRuntimeException()
+    {
+        $this->expectExceptionThrown(new \RuntimeException("Unknown batch 'unknownBatch'", 404));
+        $this->mockedEventDispatcher()->expects($this->once())->method('hasListeners')->will($this->returnValue(false));
+        $this->s()->execute('unknownBatch');
+    }
+
```
Comment (ohoareau) [inline tests/Itq/Common/Service/BatchServiceTest.php]: same.

File: tests/Itq/Common/Service/JsonServiceTest.php [hunk 11]
```diff
@@ -11,6 +11,8 @@
 
 namespace Tests\Itq\Common\Service;
 
+use function GuzzleHttp\Promise\is_fulfilled;
```
Comment (ohoareau) [inline tests/Itq/Common/Service/JsonServiceTest.php]: why this import ?

File: tests/Itq/Common/Service/JsonServiceTest.php [hunk 12]
```diff
@@ -21,4 +23,60 @@
...
+    /**
+     * @return JsonService
+     */
+    public function s()
+    {
+        /** @noinspection PhpIncompatibleReturnTypeInspection */
+
+        return parent::s();
+    }
+
```
Comment (ohoareau) [inline tests/Itq/Common/Service/JsonServiceTest.php]: - blank line

File: tests/Itq/Common/Service/JsonServiceTest.php [hunk 13]
```diff
@@ -21,4 +23,60 @@
...
+    /**
+     * @return array
+     */
+    public function getPhpData()
+    {
+        return [
+            'some date', 'associative' => ['key' => [1, 2, 3, 'string', ], 'key2' => 'value'],
+        ];
+    }
+
```
Comment (ohoareau) [inline tests/Itq/Common/Service/JsonServiceTest.php]: - blank line

File: tests/Itq/Common/Service/JsonServiceTest.php [hunk 14]
```diff
@@ -21,4 +23,60 @@
...
+    /**
+     * @return string
+     */
+    public function getJsonData()
+    {
+        return json_encode($this->getPhpData());
+    }
+
```
Comment (ohoareau) [inline tests/Itq/Common/Service/JsonServiceTest.php]: - blank line

File: tests/Itq/Common/Service/JsonServiceTest.php [hunk 15]
```diff
@@ -21,4 +23,60 @@
...
+    /**
+     * @group unit
+     */
+    public function testSerialize()
+    {
+        $options = [JSON_HEX_TAG, JSON_HEX_APOS];
+        $actual = $this->s()->serialize($this->getPhpData(), $options);
+        $this->assertEquals($this->getJsonData(), $actual);
+    }
+
```
Comment (ohoareau) [inline tests/Itq/Common/Service/JsonServiceTest.php]: - blank line

File: tests/Itq/Common/Service/JsonServiceTest.php [hunk 16]
```diff
@@ -21,4 +23,60 @@
...
+     * @group unit
+     */
+    public function testUnserializeWithNoStringThrowMalformedException()
+    {
+        $this->expectException('RuntimeException');
+        $this->expectExceptionMessage('Only string are JSON unserializable');
+        $this->expectExceptionCode(412);
+        $this->s()->unserialize(['a tab is not a string']);
+    }
+
```
Comment (ohoareau) [inline tests/Itq/Common/Service/JsonServiceTest.php]: - blank line

File: tests/Itq/Common/Service/MathServiceTest.php [hunk 17]
```diff
@@ -33,6 +33,7 @@ public function s()
 
         return parent::s();
     }
+
```
Comment (ohoareau) [inline tests/Itq/Common/Service/MathServiceTest.php]: same.

File: tests/Itq/Common/Service/MathServiceTest.php [hunk 18]
```diff
@@ -53,36 +54,49 @@ public function testDeduplicateSumMode()
             $this->s()->deduplicate($points, 'sum')
         );
     }
+
```
Comment (ohoareau) [inline tests/Itq/Common/Service/MathServiceTest.php]: same.

File: tests/Itq/Common/Service/MathServiceTest.php [hunk 19]
```diff
@@ -53,36 +54,49 @@ public function testDeduplicateSumMode()
...
         $this->assertEquals(
             ['2016-06-27' => 15, '2016-06-28' => 5],
-            $this->s()->deduplicate([['2016-06-27' => 10], ['2016-06-27' => 15], ['2016-06-27' => 20], ['2016-06-28' => 5]], 'median')
+            $this->s()->deduplicate(
+                [['2016-06-27' => 10], ['2016-06-27' => 15], ['2016-06-27' => 20], ['2016-06-28' => 5]],
+                'median'
+            )
         );
     }
+
```
Comment (ohoareau) [inline tests/Itq/Common/Service/MathServiceTest.php]: same.

File: tests/Itq/Common/Service/Stub/VolatileModel.php [hunk 20]
```diff
@@ -10,6 +10,9 @@
 
 namespace Tests\Itq\Common\Service\Stub;
 
+/**
+ * Class VolatileModel
```
Comment (ohoareau) [inline tests/Itq/Common/Service/Stub/VolatileModel.php]: @author instead

File: tests/Itq/Common/Service/YamlServiceTest.php [hunk 21]
```diff
@@ -74,8 +74,8 @@ public function testUnSerialize($expected, $value, $options)
     public function getUnserializeData()
     {
         return [
-            [['a' => 12],'a: 12', []],
-            [['a' => null], 'a: null', []],
+            [['a' => 12], 'a: 12', [], ],
```
Comment (ohoareau) [inline tests/Itq/Common/Service/YamlServiceTest.php]: remove trailing comma if one-liner array definition.

File: tests/Itq/Common/Service/JsonServiceTest.php (hunk 11 above)
Comment (Agi-dev) [reply tests/Itq/Common/Service/JsonServiceTest.php]: On autocompletion for json_encode, phpStorm autocomplete from Guzzle json_encode first by mistake. I delete it.

**Pull request merged**